lib/spack/spack/solver/concretize.lp

```diff
@@ -1566,6 +1566,29 @@ compiler_used_as_a_library(node(X, Child), Hash) :-
     attr("virtual_on_edge", PackageNode, CompilerNode2, "cxx"),
     CompilerNode1 != CompilerNode2.
 
+error(100, "Compiler mixing is disabled") :-
+    attr("virtual_on_edge", PN1, CompilerNode1, "c"),
+    attr("virtual_on_edge", PN2, CompilerNode2, "c"),
```

Comment:
Bit of inconsistent naming compared to the rest of the file.

`ParentNode` + `ChildNode` seems more common.